### PR TITLE
Adds ignore_index to sequence parallel cross entropy

### PR DESCRIPTION
**Pull request by ronald-d-rogers**
Simple change to add `ignore_index` to 100% integration w/ Hugging Face:
https://github.com/huggingface/transformers/pull/35301

I also considered adding `reduction` here, and just returning `loss.mean()/sum()` accordingly. I can add that as well if you all thinks that makes sense.

### CODE REVIEW

**Comment by ronald-d-rogers** [comment]
@microsoft-github-policy-service agree [company="Bloomberg Industry Group"]

**Comment by ronald-d-rogers** [comment]
@microsoft-github-policy-service agree company="Bloomberg Industry Group"

**Review by loadams — APPROVED** [review]

File: deepspeed/sequence/cross_entropy.py [hunk 1]
```diff
@@ -56,5 +60,9 @@ def backward(ctx, grad_output):
         return grad_input, None, None, None
 
 
-def vocab_sequence_parallel_cross_entropy(vocab_parallel_logits, target, sp_group):
-    return _VocabSequenceParallelCrossEntropy.apply(vocab_parallel_logits, target, sp_group)
+def vocab_sequence_parallel_cross_entropy(
```
[the remark this answers is not in the document]
Comment (ronald-d-rogers) [reply deepspeed/sequence/cross_entropy.py]: Thank you!

**Comment by ronald-d-rogers** [comment]
Hello all, after working on the integration more I no longer think this PR is needed. Sorry about that. Closing until further notice.
